interfaces/cython/cantera/examples/onedim/diffusion_flame_extinction.py

```diff
@@ -14,14 +14,19 @@
 
 import numpy as np
 import os
+from importlib.util import find_spec
 
 import cantera as ct
 import matplotlib.pyplot as plt
 
-# Create directory for output data files
-data_directory = 'diffusion_flame_extinction_data/'
-if not os.path.exists(data_directory):
-    os.makedirs(data_directory)
+hdf_output = find_spec('h5py') is not None
+
+if not hdf_output:
+    # Create directory for output data files
+    data_directory = 'diffusion_flame_extinction_data/'
+    if not os.path.exists(data_directory):
+        os.makedirs(data_directory)
```

Comment:
Same comment here about `pathlib`

Reply:
Again, not sure. I can replace this using `pathlib` but still believe that XML output should be dropped here eventually, which would render this moot.